### PR TITLE
Update Pinto query validator failed log, minor refactor pinot visibility store to remove panics

### PR DESCRIPTION
**Pull request by neil-xie**
**What changed?**
Update Pinto query validator failed log to include more details
Refactor pinot visibility store code to remove panics

**Why?**

**How did you test it?**
Unit test 

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/persistence/pinot/pinotVisibilityStore.go [hunk 1]
```diff
@@ -770,20 +797,21 @@ func (v *pinotVisibilityStore) getListWorkflowExecutionsByQueryQuery(tableName s
...
-		return query.String()
+		return query.String(), nil
 	}
 
 	comparExpr, orderBy := parseOrderBy(requestQuery)
 	comparExpr, err = v.pinotQueryValidator.ValidateQuery(comparExpr)
 	if err != nil {
-		v.logger.Error(fmt.Sprintf("pinot query validator error: %s", err))
+		v.logger.Error(fmt.Sprintf("pinot query validator error: %s, query: %s", err, request.Query))
+		return "", err
```
Comment (3vilhamster) [inline common/persistence/pinot/pinotVisibilityStore.go:814]:
```suggestion
		return "", fmt.Errorf("query validation: %w, query %s", err, request.Query)
```

Then you can avoid logging inside a component and only log failure higher by the stack.

File: common/persistence/pinot/pinotVisibilityStore.go [hunk 2]
```diff
@@ -907,14 +935,14 @@ func findLastOrderBy(list []string) int {
...
+func getListWorkflowExecutionsQuery(tableName string, request *p.InternalListWorkflowExecutionsRequest, isClosed bool) (string, error) {
 	if request == nil {
-		return ""
+		return "", nil
 	}
 
 	token, err := pnt.GetNextPageToken(request.NextPageToken)
 	if err != nil {
-		panic(fmt.Sprintf("deserialize next page token error: %s", err))
+		return "", err
```
Comment (3vilhamster) [inline common/persistence/pinot/pinotVisibilityStore.go:945]:
```suggestion
		return "", fmt.Errorf("next page token: %w", err)
```

File: common/persistence/pinot/pinotVisibilityStore.go (hunk 1 above)
Comment (3vilhamster) [reply common/persistence/pinot/pinotVisibilityStore.go:814]: Read more on https://github.com/uber-go/guide/blob/master/style.md#error-wrapping and https://github.com/uber-go/guide/blob/master/style.md#handle-errors-once

File: common/persistence/pinot/pinotVisibilityStore.go [hunk 3]
```diff
@@ -968,19 +996,19 @@ func getListWorkflowExecutionsByTypeQuery(tableName string, request *p.InternalL
...
```
Comment (3vilhamster) [inline common/persistence/pinot/pinotVisibilityStore.go:999]: error wrapping

File: common/persistence/pinot/pinotVisibilityStore.go [hunk 4]
```diff
@@ -1004,19 +1032,19 @@ func getListWorkflowExecutionsByWorkflowIDQuery(tableName string, request *p.Int
...
```
Comment (3vilhamster) [inline common/persistence/pinot/pinotVisibilityStore.go:1035]: error wrapping

File: common/persistence/pinot/pinotVisibilityStore.go [hunk 5]
```diff
@@ -1047,14 +1075,14 @@ func getListWorkflowExecutionsByStatusQuery(tableName string, request *p.Interna
...
```
Comment (3vilhamster) [inline common/persistence/pinot/pinotVisibilityStore.go:1078]: error wrapping

File: common/persistence/pinot/pinotVisibilityStore.go (hunk 1 above)
Comment (neil-xie) [reply common/persistence/pinot/pinotVisibilityStore.go:814]: Thanks, great to know these, will update them!

File: common/persistence/pinot/pinotVisibilityStore.go [hunk 6]
```diff
@@ -749,14 +776,14 @@ func (v *pinotVisibilityStore) getCountWorkflowExecutionsQuery(tableName string,
...
+func (v *pinotVisibilityStore) getListWorkflowExecutionsByQueryQuery(tableName string, request *p.ListWorkflowExecutionsByQueryRequest) (string, error) {
 	if request == nil {
-		return ""
+		return "", nil
 	}
 
 	token, err := pnt.GetNextPageToken(request.NextPageToken)
 	if err != nil {
-		panic(fmt.Sprintf("deserialize next page token error: %s", err))
+		return "", fmt.Errorf("next page token: %w", err)
```
Comment (davidporter-id-au) [inline common/persistence/pinot/pinotVisibilityStore.go:786]: Nit: What about logging the query just before returning the error, else that info will be lost?

**Review by davidporter-id-au — APPROVED** [review]

**Review by taylanisikdemir — APPROVED** [review]

File: common/persistence/pinot/pinotVisibilityStore.go (hunk 6 above)
Comment (neil-xie) [reply common/persistence/pinot/pinotVisibilityStore.go:795]: Sure, added log in the outer method that calls this get query method.

**Pull request merged**